boot/bootstate20.go

```diff
@@ -284,7 +284,7 @@ func (ks20 *bootState20Kernel) revisions() (curSnap, trySnap snap.PlaceInfo, try
 	tryKernel, err := ks20.bks.tryKernel()
 	// if err is ErrNoTryKernelRef, then we will just return nil as the trySnap
 	if err != nil && err != bootloader.ErrNoTryKernelRef {
-		return kern, nil, "", newTrySnapErrorf("cannot identify try kernel snap: %v", err)
+		return kern, nil, status, newTrySnapErrorf("cannot identify try kernel snap: %v", err)
```

Comment:
if this matter, we should have a test where this makes a difference, changing it back doesn't change any test outcome atm

Reply:
also will not this actually introduce new ways to have reboot loops?

Reply:
I've added a test - no, hopefully this will not introduce new reboot loops. This will make sure we reboot if the status is unexpected when we are trying ("try" instead of "trying" for kernel). The bootloader will change always the status in the next boot so in theory there should not be a loop.